### PR TITLE
Miscellaneous CI cleanup and fixes.

### PR DESCRIPTION
**Pull request by quyykk**
Some misc CI cleanup:

- Updated the vcpkg dependencies
- Prevent partial saves when CD runs are cancelled/fail
- Move various jobs from ci.yml to projects_check.yml
- Make the step names a bit more consistent
- Make the job names a bit more consistent
- Actually trigger the data parsing test if only the script was changed
- Separate the DLLs from the binary file artifact on Windows (this is so that people can download just the binary)

## Testing Done

CI will test everything.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,45 +1,24 @@
...
-# - Construct even more intelligent workflows
-#    e.g.
-#      data/* -> use "latest" build for triggering branch, only run parse and/or acceptance tests
-#      source/* -> build new binary, run all tests, cache binary
-#      PR opened -> upload artifact ZIPs to PR as comment, and edit/replace as new commits come in
-# - Windows: Allow programmers to use their own copy of development libs without hacking a temporary URL in the windows step
-
 on:
   push:
-    # Run for pushes to master or a release branch, e.g. a PR was merged ...
```
Comment (MCOfficer) [inline .github/workflows/ci.yml:19]: Why remove these? same for the other lines

Comment (quyykk) [reply .github/workflows/ci.yml:19]: oops collateral damage

**Review by samrocketman — APPROVED** [review]

**Comment by samrocketman** [comment]
I take particular interest in this change because upgrading vcpkg fixes sound for some Linux environments such as docker dev containers.

File: .github/path-filters.yml [hunk 2]
```diff
@@ -22,9 +16,6 @@ unit_tests:
   - 'tests/unit/**/*.cpp'
 
 integration_tests:
-  - 'utils/test_parse.*'
-  - 'tests/integration/run_tests*'
-  - 'data/tests/**'
```
[the remark this answers is not in the document]
Comment (quyykk) [reply .github/path-filters.yml:27]: 1. The test parse scripts are run to check for data parsing errors, they're not used for integration tests.
2. Those scripts aren't used anymore for CI
3. The location for the integration tests was moved, there is nothing on this path anymore

**Review by MCOfficer — APPROVED** [review]

**Pull request merged**